File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
       set<char> se;
        int n=s.size();
        int i=0,j=0;
        int ans=0;
        while(i<n && j<n){
            if(se.count(s[i])==0){
                se.insert(s[i]);
                i++;
                ans= max(ans,(int)se.size());
            }
            else{
                se.erase(s[j]);
                j++;
            }
        }
        return ans;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        seen = set()
        res,i,j,sz=0,0,0,0
        while i<=j and j<len(s):
            if i==j:
                seen.add(s[j])
                j+=1
                res=max(res,1)
                continue
            if s[j] in seen:
                seen.remove(s[i])
                i+=1
            else:
                seen.add(s[j])
                j+=1
            res=max(res,len(seen))
        return res
            